crates/oxc_linter/src/rules/import/no_named_as_default_member.rs

```diff
@@ -0,0 +1,175 @@
+    /// import bar from './bar'
+    /// const bar = foo.bar // trying to access named export via default
+    /// ```
+    NoNamedAsDefaultMember,
+    nursery
+);
+
+impl Rule for NoNamedAsDefaultMember {
+    fn run_once(&self, ctx: &LintContext<'_>) {
+        let module_record = ctx.semantic().module_record();
+
+        let mut has_members_map: HashMap<&Atom, (Ref<'_, Atom, _, _>, Atom)> = HashMap::default();
+        for import_entry in &module_record.import_entries {
+            let ImportImportName::Default(_) = import_entry.import_name else {
+                continue;
+            };
+
+            let specifier = import_entry.module_request.name();
+            let Some(remote_module_record_ref) = module_record.loaded_modules.get(specifier) else {
+                continue;
+            };
+
+            if !remote_module_record_ref.exported_bindings.is_empty() {
+                has_members_map.insert(
+                    import_entry.local_name.name(),
+                    (remote_module_record_ref, import_entry.module_request.name().to_owned()),
+                );
+            }
+        }
+
+        if has_members_map.is_empty() {
+            return;
+        };
+        let get_external_module_name_if_has_entry = |module_name: &Atom, entry_name: &Atom| {
+            has_members_map.get(&module_name).and_then(|it| {
+                if it.0.exported_bindings.contains_key(entry_name) {
+                    Some(it.1.to_string())
+                } else {
+                    None
+                }
+            })
+        };
+
+        let process_member_expr = |member_expr: &MemberExpression| {
+            let Expression::Identifier(ident) = member_expr.object() else {
+                return;
+            };
+            let Some(prop_str) = member_expr.static_property_name() else {
+                return;
+            };
+            if let Some(module_name) =
+                get_external_module_name_if_has_entry(&ident.name, &Atom::new_inline(prop_str))
+            {
+                ctx.diagnostic(NoNamedAsDefaultMemberDignostic(
+                    match member_expr {
+                        MemberExpression::ComputedMemberExpression(it) => it.span,
+                        MemberExpression::StaticMemberExpression(it) => it.span,
+                        MemberExpression::PrivateFieldExpression(it) => it.span,
+                    },
+                    ident.name.to_string(),
+                    prop_str.to_string(),
+                    module_name,
+                ));
+            };
+        };
+
+        for item in ctx.semantic().nodes().iter() {
+            match item.kind() {
+                AstKind::MemberExpression(member_expr) => process_member_expr(member_expr),
+                AstKind::VariableDeclarator(decl) => {
+                    if let Some(Expression::MemberExpression(member_expr)) = &decl.init {
+                        process_member_expr(member_expr);
+                        return;
```

Comment:
@XantreGodlike I know why, these `return` should be `continue`? 🙃 Change this will let the below case fail. 

```js
import baz from './named-exports';
  {
      const baz = {};
      const a = baz.a;
  }
```

Reply:
Agree, will fix it in this week

Reply:
#2071